language/stdlib/modules/DiemAccount.move

```diff
@@ -1627,6 +1638,24 @@ module DiemAccount {
             Errors::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY),
         );
 
+
+        let num_secondary_signers = Vector::length(&secondary_addresses);
+        assert(
+            Vector::length(&secondary_public_keys) == num_secondary_signers,
+            Errors::invalid_argument(PROLOGUE_EBAD_SECONDARY_PUB_KEYS),
```

Comment:
Instead of `EBAD_`, it would be nice if the name could evoke that the problem is a mismatch between the # of keys and # of signers addresses.

Reply:
You'll also need to update the logic in Rust that handles these errors and converts them specific transaction discard errors (convoluted...I know lol) so that it can handle this error correctly. Pointers: 
* https://github.com/diem/diem/blob/master/language/diem-vm/src/errors.rs#L26
* https://github.com/diem/diem/blob/master/language/diem-vm/src/errors.rs#L41

And a new prologue discard error for this should probably be added to the set of VM status codes here:
* https://github.com/diem/diem/blob/master/language/move-core/types/src/vm_status.rs#L467